what-is-pair.org

```diff
@@ -0,0 +1,253 @@
+each versioned, and if there is an update to one of them, without a coordinate
+update to the rest of them, then it is likely emacs will not work properly.
+Because of this, we've been conservative with updates to our custom
+configuration.
+
+The biggest difference in our configuration is the support of tmate blocks in org-mode,
+and to have these blocks open up appropriate windows in the Environment tmate session. This
+was developed so we could write executable documents in cluster, and see the output in our
+shared tmate session.
+
+* Comparisons to other solutions
+** Kind and Minikube
+Kind and Minikube are two ways to quickly run kubernetes clusters, and are often
+recommended for trying out kubernetes or trying out some new tool for kubernetes.
+
+You can compare them to Pair as all offer a simple way to create a cluster relatively quickly.
+
+The biggest differences are:
+- Pair is a production level cluster, running on bare metal equinix servers.  Kind and minikube
+  run inside docker and a virtual box respectively.  They cannot simulate a production-grade environment.
+- Pair can be easily setup as a multi-node cluster.  This is impossible with Minikube and difficult with Kind.
+- Pair offers built-in, simple DNS and name management. Neither come by default in Kind or Minikube. It is much
+  easier to access your Pair Environment and Pair cluster remotely.
+- Pair comes with an in-cluster coding environment.
+- Pair has a method for deploying services to yoru cluster given nothing but the repository link (assuming the repo has
+  a proper .sharing.io/init file)
+- Pair has a web interface for creating new clusters. Kind and Minikube are just cli.
+** Gitpod
+Gitpod creates remote, browser-based development environments that are configured based
+on a github repo.  Given a repo, it sets up a browser version of VSCode with necessary language tools
+and other configurations added.
+
+It's chief difference with Pair is one of focus.
+- Gitpod creates a custom development environment that happens to exist in a
+  container (possibly) on a
+cluster and focused on the needs of a single repository.
+- Pair creates a custom cluster, with multiple services added from multiple
+  repos, and happens to have a custom development environment that comes with
+  it.
+- Gitpod focuses on VSCode, but offers basic support for vim/emacs. Pair's
+  Environment focusses on Emacs, with no current support for any non-terminal
+  editors.
+- Gitpod is designed to give a contained, safe, reproducible space for new
+  developers. Pair offers an open and changing space for system administrators,
+  dev ops, and developers.
+** Coder
+Coder is similar to Gitpod, where its focus is to create custom development
+environments. Coder gives more customization options, like the computing power of
+the environment and additional processes one needs running. It is designed for
+enterprise, where you have a codebase you want to remain secure (and not on
```

Comment:
Is it designed for the enterprise?

Reply:
I think so.  It can of course be used outside of that, but many of its core features have the biggest advantage for enterprise customers--keeping proprietary code secure behind custom firewalls,  having that code only live on your own infra instead of any developer's local computer, and having new hires being able to commit to this codebase on day one all feel like chiefly enterprise benefits.

Reply:
I think the key thing here is easy onboarding to cloud native collaborative development environments to increase contribution:

> Development environment setup is something I see way too often missing in contributor files. It is super important because, if people want to contribute, they need to set up their environment.

https://twitter.com/fuzzychef/status/1560036308772274176